Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm

```diff
@@ -191,16 +184,9 @@ - (instancetype)initWithCoder:(NSCoder *)coder
     if (!selfPtr)
         return nil;
 
-    BOOL hasBaseURL;
-    [coder decodeValueOfObjCType:"c" at:&hasBaseURL size:sizeof(hasBaseURL)];
-
-    RetainPtr<NSURL> baseURL;
-    if (hasBaseURL)
-        baseURL = (NSURL *)[coder decodeObjectOfClass:NSURL.class forKey:baseURLKey];
-
     NSUInteger length;
     if (auto bytes = (UInt8 *)[coder decodeBytesWithReturnedLength:&length]) {
-        m_wrappedURL = bridge_cast(adoptCF(CFURLCreateAbsoluteURLWithBytes(nullptr, bytes, length, kCFStringEncodingUTF8, bridge_cast(baseURL.get()), true)));
+        m_wrappedURL = bridge_cast(adoptCF(CFURLCreateAbsoluteURLWithBytes(nullptr, bytes, length, kCFStringEncodingUTF8, nullptr, true)));
```

Comment:
Could it be that this is wrong? Maybe it needs to be using the baseURL to create the true URL?

Reply:
I looked through the CFNetwork API, and don't see a Create function that takes baseURL (maybe I'm missing something).

In that case, the special handling of baseURL is silly, and maybe we can do away with the `WKSecureCodingURLWrapper` class entirely?

Reply:
This is not what was wrong. Decoding the baseURL would throw a few lines above in this function.

Why do we want to do away with WKSecureCodingURLWrapper? I thought this was a convenient way to code/decoder a URL as bytes.

Reply:
> Could it be that this is wrong? Maybe it needs to be using the baseURL to create the true URL?

We use to pass a baseURL because the `bytes` only contained the relative part of the URL. We needed both the baseURL + related part bytes to construct the absolute URL.

Now though, I encode the absolute URL as the bytes. As a result, we don't need to pass a base URL. Note that this is covered by my new API test.

Reply:
Ah, very nice.,